src/nbl/video/ILogicalDevice.cpp

```diff
@@ -834,12 +845,10 @@ bool ILogicalDevice::createGraphicsPipelines(
 )
 {
     std::fill_n(output, params.size(), nullptr);
-    IGPUGraphicsPipeline::SCreationParams::SSpecializationValidationResult specConstantValidation = commonCreatePipelines(nullptr, params,
-        [this](const asset::IPipelineBase::SShaderSpecInfo& info)->bool
+    SSpecializationValidationResult specConstantValidation = commonCreatePipelines(nullptr, params,
+        [this](const IGPUPipelineBase::SShaderSpecInfo& info)->bool
         {
-            if (info.stage != hlsl::ShaderStage::ESS_VERTEX)
-                return true;
-            return info.shader;
+            return info.shader != nullptr;
```

Comment:
this is already being checked before calling `extra` lambda, you now have a problem we're not checking the stage presence mask for `VERTEX` being non-conditionally present.

You need to add a parameter to `commonCreatePipelines` which is `requiredStagesMask` (Vertex for graphics, raygen for RT, compute for compute) and then this lambda can always return true

Reply:
basically looking at the different `createPipelines` there's no need for a lambda, the function can just check extra stuff for `requiresSubgroupSize` and `requiredSubgroupSizeStages` (whenever a size is required, the stage must be in the physical device limits of what stages are allowd to require a subgroup size)